### PR TITLE
Reworked the StatTracker client-side

### PR DESCRIPTION
**Pull request by Celant**
Reworked the StatTracker client side stuff to match the new server side
changes. Also fixed the memory tracking, plus added a 'providerid' for
certain server hosting providers

### CODE REVIEW

**Comment by WhiteXZ** [comment]
@nicatronTg @Ijwu @tylerjwatson

**Comment by nicatronTg** [comment]
lgtm

**Comment by Ijwu** [comment]
-insert LGTM gif here-

File: TShockAPI/StatTracker.cs [hunk 1]
```diff
@@ -7,13 +7,24 @@
...
 using System.Web;
 using TerrariaApi.Server;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
 
 namespace TShockAPI
 {
 	public class StatTracker
 	{
+		[DllImport("kernel32.dll")]
```
Comment (nicatronTg) [inline TShockAPI/StatTracker.cs]: My only question is: what happens when this executes on mono?

Comment (Celant) [inline TShockAPI/StatTracker.cs]: Hmm, I thought I tested that on mono, unless it was before I added that part. If someone has a mono setup ready to go, can they test it? As I don't :(

Comment (Celant) [inline TShockAPI/StatTracker.cs]: The answer is, nothing. It apparently just skips over this with no exception thrown and continues normally.

**Comment by Celant** [comment]
@nicatronTg Opt-out added and things final checked. If I could get your verdict that'd be great :)

**Comment by nicatronTg** [comment]
@Celant make the command line argument more descriptive? --stats-optout?

**Comment by Celant** [comment]
@nicatronTg Suggested changes have been made

**Pull request merged**